ui/src/source.ts

```diff
@@ -127,6 +127,15 @@ export const object = objectStore.readable;
 const revisionsStore = remote.createStore<Revisions>();
 export const revisions = revisionsStore.readable;
 
+export const objectType = writable(ObjectType.Tree);
```

Comment:
I think these actions would make sense as events in the event loop, so we can keep things consistent. If that's a lot for this PR, we should make a followup issue. Having both approaches (some updates use the event structure, others don't) adds some unnecessary complexity, so I feel we should go with one or the other.

Reply:
On point observation.

Reply:
> I think these actions would make sense as events in the event loop, so we can keep things consistent. If that's a lot for this PR, we should make a followup issue. Having both approaches (some updates use the event structure, others don't) adds some unnecessary complexity, so I feel we should go with one or the other.

I am not sure what you mean with `event in the event loop`. We set up an infrastructure for remote stores, ie stores where we have a loading, error etc. state. The stores added here are simpler, so setting and reading them is actually enough. For me using them as remote stores would add more complexity.

Reply:
Have to agree with @MeBrei here, I don't see what wrapping everything in the event code would add here besides another layer of indirection.

Reply:
Cool, we can discuss elsewhere